### PR TITLE
Slow down failed login attempts

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: doc/en/user/source/security/webadmin/auth.rst [hunk 1]
```diff
@@ -5,6 +5,74 @@ Authentication
 
 This page manages the authentication options, including authentication providers and the authentication chain.
    
+Brute force attack prevention
+-----------------------------
+
+GeoServer ships with a simple delay based brute force attack prevention system.
```
Comment (jodygarnett) [inline doc/en/user/source/security/webadmin/auth.rst]: No neee to use the word simple

File: doc/en/user/source/security/webadmin/auth.rst [hunk 2]
```diff
@@ -5,6 +5,74 @@ Authentication
...
+     - Whether the brute force attack prevention is enabled. Defaults to true.
+   * - Minimum delay on failed authentication (seconds)
+     - Minimum number of seconds a failed login request will be made to wait before getting a response
+   * - Maximum delay on failed authentication (seconds)
+     - Maximum number of seconds a failed login request will be made to wait before getting a response
+   * - Excluded network masks
+     - Network masks identifying hosts that are excluded from brute force attack prevention. Can be empty, include specific IPs, or a list of network masks. 
+       Defaults to 127.0.0.1, the localhost.
+   * - Maximum number of threads blocked on failed login delay
+     - Limits the number of threads that get delayed on failed login, should be set to a value less than the container's available response threads.
```
Comment (jodygarnett) [inline doc/en/user/source/security/webadmin/auth.rst]: Can we determine that vale at runtime and provide a warning or something?

File: doc/en/user/source/security/webadmin/auth.rst [hunk 3]
```diff
@@ -5,6 +5,74 @@ Authentication
...
+       
+The mechanism works as follows:
```
Comment (jodygarnett) [inline doc/en/user/source/security/webadmin/auth.rst]: Thanks explanation is clear.

File: doc/en/user/source/security/webadmin/auth.rst (hunk 2 above)
Comment (aaime) [inline doc/en/user/source/security/webadmin/auth.rst]: Nope, no way to know how large the servlet container thread pool is

**Comment by aaime** [comment]
Yep, I know that "brute force" is the way, I just wanted to avoid drawing too much attention on the pull request :-p

**Comment by aaime** [comment]
Merged on master, will wait tomorrow to get the build server check before backporting the changes.

**Pull request merged**